src/core/zowe/core_for_zowe_sdk/session.py

```diff
@@ -0,0 +1,71 @@
+"""Zowe Python Client SDK.
+
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+
+https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zowe Project.
+"""
+
+from dataclasses import dataclass
+from typing import Union
+
+from zowe.core_for_zowe_sdk.zosmf_profile2 import ProfileManager
+from . import session_constants
+
+
+@dataclass
+class ISession:
+    """
+    Class to represent session parameters
+    """
+
+    hostname: str
+    port: int
+    rejectUnauthorised: bool = True
+    user: Union[str, None] = None
+    password: Union[str, None] = None
+    protocol: Union[str, None] = None
+    basePath: Union[str, None] = None
+    type: Union[str, None] = None
+    base64EncodeAuth: Union[str, None] = None
+    tokenType: Union[str, None] = None
+    tokenValue: Union[str, None] = None
+
+
+class Session:
+    """
+    Class used to represent connection details
+    """
+
+    def __init__(self, props: dict) -> None:
+        # set hostname and port
+        if props["host"] is not None and props["port"] is not None:
+            self.session: ISession = ISession(
+                hostname=props["host"], port=props["port"]
+            )
+        else:
+            raise "Hostname and Port must be supplied"
+
+        # determine authentication type
+        if props["user"] is not None and props["password"] is not None:
+            self.session.user = props["user"]
+            self.session.password = props["password"]
+            self.session.rejectUnauthorised = props["rejectUnauthorised"]
```

Comment:
The "rejectUnauthorized" property may be set for any authentication type and may not always be defined in the `props` dictionary.

To load the "rejectUnauthorized" value, I would suggest adding an `if` condition outside of the user/password one. Something like the following:
```
if props.get("rejectUnauthorized") is not None:
    self.session.rejectUnauthorized = props["rejectUnauthorized"]
```